File: backend/Entities/Dtos/ExpenseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Dtos
{
    public class ExpenseDto
    {
        public int ExpenseId { get; set; }
        public int ExpenseCategoryId { get; set; }
        public string ExpenseCategoryName { get; set; }
        public string Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime ExpenseDate { get; set; }
        public string Notes { get; set; }
        public DateTime CreatedDate { get; set; }
    }
}

File: backend/frontend/src/app/models/expense.dto.ts
export interface ExpenseCategoryDto {
    expenseCategoryId: number;
    name: string;
    description: string;
}

export interface ExpenseDto {
    expenseId: number;
    expenseCategoryId: number;
    expenseCategoryName: string;
    description: string;
    amount: number;
    expenseDate: Date;
    notes: string;
    createdDate: Date;
}

export interface ExpenseAddDto {
    expenseCategoryId: number | null;
    description: string;
    amount: number;
    expenseDate: string; // Tarihi ISO formatında göndermek daha güvenilirdir
    notes?: string;
}

export interface ExpenseUpdateDto {
    expenseId: number;
    expenseCategoryId: number;
    description: string;
    amount: number;
    expenseDate: string;
    notes?: string;
}